src/MainWindow.h

```diff
@@ -0,0 +1,257 @@
+     * If a row is selected, the bill corresponding to that row will have its funding status set to Funded.
+     */
+    void fundBillOnClick();
+
+    /**
+     * @brief Called when the Defund Bill button is pressed. Iterates over the rows and columns of the bill table widget checking for which rows are selected.
+     * If a row is selected, the bill corresponding to that row will have its funding status set to Not Funded.
+     */
+    void defundBillOnClick();
+
+    /**
+     * @brief Called when the Delete Bill button is pressed. Iterates over the rows and columns of the bill table widget checking for which rows are selected.
+     * If a row is selected, the bill corresponding to that row will have its funding status checked for a value of Funded.
+     * If the bill is funded, the amount of money allocated to that bill is returned to the total amount available, the bill is removed from the funded bills list, and lastly the row is removed from the bill table widget.
+     */
+    void deleteBillOnClick();
+
+private:
+
+    /**
+     * @brief Creates a message box in the event that the config file was unable to be generated in the desired directory before terminating the application.
+     */
+    void createConfigFileGenerateFailureBox();
+
+    /**
+     * @brief Creates a message box in the event that the attempting to open the config file resulted in an error.
+     *  If the user chooses to press the "Ok" button, the welcome sequence will initiate where the user will provide their financial information again.
+     *  If the user chooses to press the "Close" button, the application will terminate.
+     */
+    void createCorruptConfigFileBox();
+
+    /**
+     * @brief Creates a message box in the event that an invalid key was found in the config file.
+     * Warns the user to correct the config file, before terminating the application.
+     */
+    void createInvalidKeyBox();
+
+    /**
+     * @brief Creates a message box in the event that a user attemps to save a bill they entered which is missing a name or an amount due.
+     */
+    void createMissingBillDetailsBox();
+
+    /**
+     * @brief Checks if the pre-determined directory for the config file can be generated, if it cannot an error message is displayed to the user. Then checks if the config file exists in that directory, if so the file is parsed.
+     * If the file does not exist in the directory, the welcome sequence initiates in which the user is asked for their financial information.
+     */
+    void attemptConfigFileGeneration();
+    /**
+     * @brief Generates a message box to welcome a first time user.
+     * Has clickable buttons allowing the user to advance to the total funds available prompt, and to exit the application.
+     */
+    void welcomeFirstTimeUser();
+
+    /**
+     * @brief Asks the user for the total amountof money they have available for spending.
+     * Saves the input to a member variable for usage.
```

Comment:
*for usage* is vague, either describe what it will be used for or just remark that it is saved.

Reply:
Further clarification added.